notification-server/src/bin/api.rs

```diff
@@ -51,45 +91,75 @@ lazy_static! {
     static ref MAX_PREFERENCES_LENGTH: usize = all::<Preference>().collect::<Vec<_>>().len();
 }
 
-#[tracing::instrument(skip(state))]
-async fn upsert_account_device(
-    Path(device): Path<String>,
-    State(state): State<Arc<AppState>>,
-    Json(subscription): Json<DeviceSubscription>,
-) -> Result<impl IntoResponse, Response> {
-    info!(
-        "Subscribing accounts {:?} to device {}",
-        subscription, device
-    );
+async fn handle(
```

Comment:
A more elaborate name would be nice here and possibly a short description. Maybe 
```suggestion
/// Validates and updates the subscription for a given device
async fn handle_subscription(
```
?